common/meta/types/proto/meta.proto

```diff
@@ -48,6 +48,65 @@ message ExportedChunk {
   repeated string data = 10;
 }
 
+message WatchRequest {
+  // request_union is a request to either create a new watcher or cancel an existing watcher.
+  oneof request_union {
+    WatchCreateRequest create_request = 1;
+    WatchCancelRequest cancel_request = 2;
+  }
+}
+
+message WatchCreateRequest {
+  // key is the key to register for watching.
+  string key = 1;
+
+  // range_end is the end of the range [key, range_end) to watch. If range_end is not given,
+  // only the key argument is watched. If range_end is equal to '\0', all keys greater than
+  // or equal to the key argument are watched.
+  // If the range_end is one bit larger than the given key,
+  // then all keys with the prefix (the given key) will be watched.
+  string key_end = 2;
+}
+
+message WatchCancelRequest {
+  int64 watcher_id = 1;
+}
+
+message Event {
+  enum EventType {
+    UpInsert = 0;
+    Delete = 1;
+  }
+
+  string key = 1;
+
+  EventType event = 2;
+
+  // current hold current value of key
+  bytes current = 3;
+
+  // current hold prev value of key(if any)
+  bytes prev = 4;
```

Comment:
It seems it is possible to make it an `Option`:
https://github.com/tokio-rs/prost#field-modifiers